.github/macos-build.sh

```diff
@@ -49,6 +49,6 @@ sudo make install
 cd ../..
 
 python -m pip install --upgrade pip
-python -m pip install cibuildwheel==2.21.3
+python -m pip install cibuildwheel==2.23.3
```

Comment:
A workflow that was run on a repository I own failed.
Although the details are unclear, the issue was resolved by updating the version.